### DISCUSSION TITLE
Zoom feature with access to zoom value

### DISCUSSION THREAD

**Discussion by dellyn** [idiotWu/smooth-scrollbar#420] [Q&A] [answered]
What's the best approach (How to add) zoom feature to the smooth scrollbar library?
I've been trying to implement this with transform: scale, but I have to calculate a huge chunk of data to get the right scale using transform: translate(), scale() and scrollTo() without animation

At the same time my scrollbar container width changes depending on the state of the application and it affects the position of the scrollbar content and scroll if I have zoom. Thank you in advance
#react

**Comment by dolphin-wood** [comment, marked as answer]
Zooming is quite a difficult feature that comes with a lot of re-calculations. The following link is an (inefficient) approach I tried in a recent project, implementing a scalable image viewer in [keyframe refiner](https://keyframe-refiner.js.org/). You can drag&drop an image into it and zoom it with trackpads (or holding <kbd>ctrl</kbd> and scrolling the mouse wheel) to see how it behaves.

Source: [keyframe-refiner/src/app/views/ImageViewer.svelte#L113-L146](https://github.com/keyframe-refiner/keyframe-refiner/blob/5497fe451411c612a4b0da501528b0c7e65f7106/src/app/views/ImageViewer.svelte?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L113-L146)
